lib/profileAgents.js

```diff
@@ -45,18 +45,35 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
       unique: true,
       background: false
     }
+  }, {
+    collection: 'profile-profileAgent',
+    fields: {'profileAgent.token': 1},
+    options: {
+      partialFilterExpression: {'profileAgent.token': {$exists: true}},
+      unique: true,
+      background: false
+    }
   }]);
 });
 
 /**
  * Creates a ProfileAgent and associates it with an account if provided.
  *
- * @param {String} accountId the id of the account to associate with the
- *                           ProfileAgent.
+ * @param {string} accountId - The ID of an account.
+ * @param {string} profileId - The ID of a profile.
+ * @param {string} [token] - An application token.
  *
  * @return {Promise<Object>} resolves to a ProfileAgent.
  */
-exports.create = async ({accountId, profileId} = {}) => {
+exports.create = async ({accountId, profileId, token} = {}) => {
+  // FIXME: is accountId required or optional? The api description implies that
+  // it may be, but the jsdoc does not have the expected brackets
+  assert.string(accountId, 'accountId');
```

Comment:
It's optional -- profile agents can be claimed later. We may want to prevent them from being "claimed", however, if they have a token, since they should not be associated with an account on the system.